presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -745,10 +753,22 @@ public ParquetStructConverter(Type prestoType, String columnName, GroupType entr
             this.fieldIndex = fieldIndex;
 
             ImmutableList.Builder<BlockConverter> converters = ImmutableList.builder();
-            for (int i = 0; i < prestoTypeParameters.size(); i++) {
+            for (int i = 0, n = fieldTypes.size(); i < n; i++) {
                 parquet.schema.Type fieldType = fieldTypes.get(i);
                 converters.add(createConverter(prestoTypeParameters.get(i), columnName + "." + fieldType.getName(), fieldType, i));
             }
+            if (prestoTypeParameters.size() != fieldTypes.size()) {
+                log.info("Parquet column " + columnName + " field number mismatch, metastore has: "
```

Comment:
It's good to log something since we don't fail the query any more. I can make this a warning.